components/org.wso2.identity.webhook.common.event.handler/src/main/java/org/wso2/identity/webhook/common/event/handler/EventConfigManager.java

```diff
@@ -0,0 +1,129 @@
+import org.json.simple.parser.ParseException;
+import org.wso2.carbon.identity.event.IdentityEventServerException;
+import org.wso2.identity.webhook.common.event.handler.constant.Constants;
+import org.wso2.identity.webhook.common.event.handler.model.ResourceConfig;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * This class manages event-related configurations.
+ */
+public class EventConfigManager {
+
+    private volatile ResourceConfig eventSchema = null;
+    private final Object lock = new Object();
+    private static volatile EventConfigManager instance;
+
+    private EventConfigManager() {}
+
+    public static EventConfigManager getInstance() {
+
+        if (instance == null) {
+            synchronized (EventConfigManager.class) {
+                if (instance == null) {
+                    instance = new EventConfigManager();
+                }
+            }
+        }
+        return instance;
+    }
+
+    /**
+     * Retrieve the event URI.
+     *
+     * @param eventName Event name.
+     * @return Event URI string.
+     * @throws IdentityEventServerException If an error occurs.
+     */
+    public String getEventUri(String eventName) throws IdentityEventServerException {
+
+        try {
+            ResourceConfig eventConfigObject = getEventConfig(eventName);
+            if (eventConfigObject.getConfigs() != null &&
+                    eventConfigObject.getConfigs().containsKey(Constants.EVENT_CONFIG_SCHEMA_NAME_KEY)) {
+                return (String) eventConfigObject.getConfigs().get(Constants.EVENT_CONFIG_SCHEMA_NAME_KEY);
+            } else {
+                throw new IdentityEventServerException("Event schema not found in the resource event config " +
+                        "for the eventName: " + eventName);
+            }
+        } catch (ClassCastException e) {
+            throw new IdentityEventServerException("Error while casting event config at server side", e);
+        }
+    }
+
+    /**
+     * Retrieve the event config.
+     *
+     * @param eventName Event name.
+     * @return Resource config object.
+     * @throws IdentityEventServerException If an error occurs.
+     */
+    private ResourceConfig getEventConfig(String eventName) throws IdentityEventServerException {
+
+        JSONObject eventsConfigObject = (JSONObject) getEventsSchemaResourceFile().getConfigs()
+                .get(Constants.EVENT_SCHEMA_EVENTS_KEY);
+        if (eventsConfigObject != null && !eventsConfigObject.isEmpty() &&
+                eventsConfigObject.containsKey(eventName)) {
+            return new ResourceConfig((JSONObject) eventsConfigObject.get(eventName));
+        } else {
+            throw new IdentityEventServerException("Event schema not found in the resource event config " +
+                    "for the eventName: " + eventName);
```

Comment:
Better say event schema for event name not found in the  xxx file

Reply:
Addressed with https://github.com/wso2-extensions/identity-webhook-event-handlers/pull/1/commits/dc72213e0cb8089d982fddd20de1d085f01ac28a